File: editorial/chap02/python/answer_B06.py
# 입력
N = int(input())
A = list(map(int, input().split()))
Q = int(input())
L = [ None ] * Q
R = [ None ] * Q
for i in range(Q):
	L[i], R[i] = map(int, input().split())

# 당첨의 수, 꽝의 수의 누적 합을 구한다
# 配列 A が 0 番目から始まっていることに注意！
Atari = [ 0 ] * (N + 1)
Hazre = [ 0 ] * (N + 1)
for i in range(1, N+1):
	# アタリについて計算
	Atari[i] = Atari[i - 1]
	if A[i - 1] == 1:
		Atari[i] += 1
	# ハズレについて計算
	Hazre[i] = Hazre[i - 1]
	if A[i - 1] == 0:
		Hazre[i] += 1

# 질문에 답한다
for i in range(Q):
	NumAtari = Atari[R[i]] - Atari[L[i] - 1]
	NumHazre = Hazre[R[i]] - Hazre[L[i] - 1]
	if NumAtari > NumHazre:
		print("win")
	elif NumAtari == NumHazre:
		print("draw")
	else:
		print("lose")

File: editorial/chap02/cpp/answer_B06.cpp
#include <iostream>
using namespace std;
 
int N, A[100009];
int Q, L[100009], R[100009];
int Atari[100009], Hazre[100009];
 
int main() {
	// 입력
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
	cin >> Q;
	for (int i = 1; i <= Q; i++) cin >> L[i] >> R[i];
 
	// 당첨의 수, 꽝의 수의 누적 합을 구한다
	Atari[0] = 0;
	Hazre[0] = 0;
	for (int i = 1; i <= N; i++) {
		Atari[i] = Atari[i - 1]; if (A[i] == 1) Atari[i] += 1;
		Hazre[i] = Hazre[i - 1]; if (A[i] == 0) Hazre[i] += 1;
	}
 
	// 질문에 답한다
	for (int i = 1; i <= Q; i++) {
		int NumAtari = Atari[R[i]] - Atari[L[i] - 1];
		int NumHazre = Hazre[R[i]] - Hazre[L[i] - 1];
		if (NumAtari > NumHazre) cout << "win" << endl;
		else if (NumAtari == NumHazre) cout << "draw" << endl;
		else cout << "lose" << endl;
	}
	return 0;
}
